### ISSUE TITLE
1.5.0-beta13. Android: problems with pressKeyCode and longPressKeyCode are here yet

### ISSUE THREAD

**Issue by TikhomirovSergey** [appium/appium#6112]
Server: 1.5.0-beta13

Client code samples:

```java
    AppiumServiceBuilder builder = AppiumDriverLocalService.buildDefaultService();
    service.start();

    File appDir = new File("src/test/java/io/appium/java_client");
    File app = new File(appDir, "ApiDemos-debug.apk");
    DesiredCapabilities capabilities = new DesiredCapabilities();
    capabilities.setCapability(MobileCapabilityType.BROWSER_NAME, "");
    capabilities.setCapability(MobileCapabilityType.DEVICE_NAME, "Android Emulator");
    capabilities.setCapability(MobileCapabilityType.APP, app.getAbsolutePath());
    capabilities.setCapability(MobileCapabilityType.NEW_COMMAND_TIMEOUT, 120);
    driver = new AndroidDriver<WebElement>(service.getUrl(), capabilities);

    ...
    driver.pressKeyCode(AndroidKeyCode.HOME);
    ...
    driver.longPressKeyCode(AndroidKeyCode.HOME);
```

Error which is thrown on the client side at both cases:

```
org.openqa.selenium.WebDriverException: An unknown server-side error occurred while processing the command. (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 28 milliseconds
Build info: version: '2.50.0', revision: '1070ace4650453d518aeb03e7a9a36c9d264a8e7', time: '2016-01-27 10:46:18'
System info: host: '', ip: '', os.name: 'Windows 7', os.arch: 'amd64', os.version: '6.1', java.version: '1.7.0_71'
Session ID: 89844911-617b-4aaa-93aa-82b01d7ee94f
Driver info: io.appium.java_client.android.AndroidDriver
Capabilities [{newCommandTimeout=120, platformVersion=4.4.4, app=C:\appium\java-client\src\test\java\io\appium\java_client\ApiDemos-debug.apk, platform=LINUX, databaseEnabled=false, javascriptEnabled=true, platformName=Android, deviceName=emulator-5554, browserName=, webStorageEnabled=false, networkConnectionEnabled=true, desired={newCommandTimeout=120, app=C:\..\java\io\appium\java_client\ApiDemos-debug.apk, deviceName=Android Emulator, platformName=Android, browserName=}, locationContextEnabled=false, warnings={}, takesScreenshot=true}]
```

Server outputs:
[pressKeyCode](https://gist.github.com/TikhomirovSergey/8a80c01d1d614ca6ada4)
[longPressKeyCode](https://gist.github.com/TikhomirovSergey/2329c5d6cee8b8fee5f1)

Attention please:
- pressKeycode
https://gist.github.com/TikhomirovSergey/8a80c01d1d614ca6ada4#file-gistfile1-txt-L145
https://gist.github.com/TikhomirovSergey/8a80c01d1d614ca6ada4#file-gistfile1-txt-L148

- longPressKeyCode
https://gist.github.com/TikhomirovSergey/2329c5d6cee8b8fee5f1#file-long-press-key-code-L143
https://gist.github.com/TikhomirovSergey/2329c5d6cee8b8fee5f1#file-long-press-key-code-L146

Just for comparison logs of node 1.4.x:
https://gist.github.com/TikhomirovSergey/4d10f4661a86edecc63d#file-gistfile1-txt-L11
https://gist.github.com/TikhomirovSergey/b7f35933d860e803bd05#file-gistfile1-txt-L12

Same actions with metastates work fine :+1:

**Comment by TikhomirovSergey** [comment]
appium/java-client#302
appium/appium-dotnet-driver#136

**Comment by TikhomirovSergey** [comment]
https://github.com/appium/appium/issues/5937

**Comment by imurchie** [comment]
Fixed in `appium-android-driver` 1.6.6.
